rubix-spi/src/main/java/com/qubole/rubix/spi/BookKeeperFactory.java

```diff
@@ -39,23 +114,39 @@ public BookKeeperFactory(BookKeeperService.Iface bookKeeper)
     }
   }
 
-  public RetryingBookkeeperClient createBookKeeperClient(String host, Configuration conf) throws TTransportException
+  public RetryingPooledBookkeeperClient createBookKeeperClient(String host, Configuration conf) throws TTransportException
   {
     if (bookKeeper != null) {
       return new LocalBookKeeperClient(null, bookKeeper);
```

Comment:
This needs a cleanup, we can take it up separately. We should return localBookKeeper only for Localhost and for other make connection and have a single bookkeeperfactory. Currently factory is instantiated at multiple places